Source/Core/VideoCommon/RenderBase.h

```diff
@@ -242,6 +243,8 @@ class Renderer
   void BeginUIFrame();
   void EndUIFrame();
 
+  std::unique_ptr<NetPlayChatUI>& GetNetPlayChatUI();
```

Comment:
Rather than returning a reference to the unique_ptr (and adding an additional level of indirection), how about returning m_netplay_chat_ui.get(), that way callers can't modify the pointer.

Furthermore, why does instance need to be swapped during runtime anyway? Why not just create a single instance of it always, and change the callbacks as needed?

Reply:
The idea was to actually abide by https://github.com/dolphin-emu/dolphin/pull/7903#discussion_r266224270. So the UI doesn't take up memory all of the time.